climada/hazard/tc_tracks.py

```diff
@@ -521,7 +555,15 @@ def read_ibtracs_netcdf(self, provider=None, rescale_windspeeds=True, storm_id=N
                 LOGGER.info("Progress: %d%%", perc)
                 last_perc = perc
             track_ds = ibtracs_ds.sel(storm=i_track, date_time=t_msk)
-            st_penv = xr.apply_ufunc(basin_fun, track_ds.basin, vectorize=True)
+            tr_basin_penv = xr.apply_ufunc(basin_fun, track_ds.basin, vectorize=True)
+            tr_genesis_basin = track_ds.basin.values[0].astype(str).item()
+
+            # Now that the valid time steps have been selected, we discard tracks that don't fit
+            # the specified basin definitions:
+            if genesis_basin is not None and tr_genesis_basin != genesis_basin:
```

Comment:
Would another log message be useful here?
```suggestion
            if genesis_basin is not None and tr_genesis_basin != genesis_basin:
                LOGGER.info('No tracks with first valid eye position in genesis basin %s.', genesis_basin)
```

Reply:
I added a log message after the loop.